h2o-core/src/main/java/hex/AUC2.java

```diff
@@ -362,7 +368,7 @@ private void mergeOneBin( ) {
     // centers.  Same problem for sorted data.
     private int find_smallest() {
       if( _ssx == -1 ) return (_ssx = find_smallest_impl());
-      assert _ssx == find_smallest_impl();
+      //assert _ssx == find_smallest_impl();
```

Comment:
This here definitely saved time!  There may not be a need to verify this.

Reply:
This does save time when assertions are enabled (eg.: when H2O is started from the R client).

This is however not the reason for the performance improvement, all my measurements were done with disabled assertions.